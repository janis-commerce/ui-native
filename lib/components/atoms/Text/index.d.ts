import { ReactElement } from 'react';
declare type TextProps = {
    children?: ReactElement | string;
};
declare const Text: {
    ({ children, ...props }: TextProps): JSX.Element | null;
    propTypes: {
        children: import("prop-types").Validator<NonNullable<NonNullable<import("prop-types").ReactNodeLike>>>;
    };
};
export default Text;
