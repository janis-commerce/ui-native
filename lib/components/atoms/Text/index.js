import React from 'react';
import { StyleSheet, Text as TextComponent } from 'react-native';
import { node, arrayOf, oneOfType } from 'prop-types';
const Text = ({ children, ...props }) => {
    if (!children)
        return null;
    return <TextComponent style={styles.TextStyles} {...props}>{children}</TextComponent>;
};
const styles = StyleSheet.create({
    TextStyles: {
        fontSize: 16,
        fontFamily: 'Roboto'
    }
});
Text.propTypes = {
    children: oneOfType([node, arrayOf(node)]).isRequired,
};
export default Text;
