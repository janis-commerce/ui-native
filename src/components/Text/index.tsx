import React, {ReactElement} from 'react';
import {StyleSheet, Text as TextComponent} from 'react-native';

type TextProps = {
	children?: ReactElement | string;
};

const Text = ({children, ...props}: TextProps) => {
	if (!children) {
		return null;
	}

	return (
		<TextComponent style={styles.TextStyles} {...props}>
			{children}
		</TextComponent>
	);
};

const styles = StyleSheet.create({
	TextStyles: {
		fontSize: 16,
		fontFamily: 'Roboto',
	},
});

export default Text;
