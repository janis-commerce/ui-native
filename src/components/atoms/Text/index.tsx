import React, {ReactElement} from 'react';
import {
	StyleProp,
	StyleSheet,
	Text as TextComponent,
	TextProps as TextComponentProps,
	TextStyle,
} from 'react-native';
import {moderateScale, scaledForDevice} from 'scale';
import {isDevEnv} from 'utils/index';

export interface TextProps extends TextComponentProps {
	children?: ReactElement | string;
	style?: StyleProp<TextStyle>;
}

const Text = ({children, style, ...props}: TextProps) => {
	if (!children) {
		return null;
	}

	const validFontSize = scaledForDevice(13, moderateScale);

	const styles = StyleSheet.create({
		TextStyles: {
			fontSize: validFontSize,
			fontFamily: 'Roboto',
		},
	});

	// istanbul ignore next
	if (isDevEnv()) {
		console.warn('This component is going to be deprecated soon.');
	}
	return (
		<TextComponent style={[styles.TextStyles, style]} {...props}>
			{children}
		</TextComponent>
	);
};

export default Text;
