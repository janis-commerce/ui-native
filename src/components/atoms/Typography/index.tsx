import React, {ReactElement} from 'react';
import {Text, StyleProp, TextStyle, TextProps} from 'react-native';
import getStyleByTypography from './utils/getStyleByTypography';

export type TypographyType = 'display' | 'heading' | 'title' | 'label' | 'body' | 'overline';

export type TypographySize = 'large' | 'medium' | 'small';

interface TypographyProps extends TextProps {
	children?: ReactElement | string;
	style?: StyleProp<TextStyle>;
	type?: TypographyType;
	size?: TypographySize;
}

const Typography = ({children, style, type, size, ...props}: TypographyProps) => {
	if (!children) {
		return null;
	}

	const typographyStyles = getStyleByTypography(
		type as TypographyType | 'string',
		size as TypographySize | 'string'
	);

	return (
		<Text style={[style, typographyStyles.typography]} {...props}>
			{children}
		</Text>
	);
};

export default Typography;
