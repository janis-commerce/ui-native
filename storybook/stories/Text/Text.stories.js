import React from 'react';
import Text from './index';

const fontFamilies = [
	'normal',
	'notoserif',
	'sans-serif',
	'sans-serif-light',
	'sans-serif-thin',
	'sans-serif-condensed',
	'sans-serif-medium',
	'serif',
	'Roboto',
	'monospace',
];
const fontWeights = [
	'normal',
	'bold',
	'100',
	'200',
	'300',
	'400',
	'500',
	'600',
	'700',
	'800',
	'900',
];

export default {
	title: 'Text',
	argTypes: {
		fontFamily: {
			options: fontFamilies,
			control: {type: 'select'},
		},
		fontStyle: {
			options: ['normal', 'italic'],
			control: {type: 'select'},
		},
		fontWeight: {
			options: fontWeights,
			control: {type: 'select'},
		},
		textDecorationLine: {
			options: ['none', 'underline', 'line-through', 'underline line-through'],
			control: {type: 'select'},
		},
		textTransform: {
			options: ['none', 'uppercase', 'lowercase', 'capitalize'],
			control: {type: 'select'},
		},
	},
};

export const DefaultProps = ({
	textToDisplay,
	fontSize,
	color,
	fontFamily,
	fontStyle,
	fontWeight,
	letterSpacing,
	textDecorationLine,
	textTransform,
}) => (
	<Text
		fontSize={fontSize}
		color={color}
		fontFamily={fontFamily}
		fontStyle={fontStyle}
		fontWeight={fontWeight}
		letterSpacing={letterSpacing}
		textDecorationLine={textDecorationLine}
		textTransform={textTransform}>
		{textToDisplay}
	</Text>
);

DefaultProps.storyName = 'default props';

DefaultProps.args = {
	textToDisplay: 'Hola',
	fontSize: 24,
	color: '#000000',
	fontFamily: 'Roboto',
	fontStyle: 'normal',
	fontWeight: 'normal',
	letterSpacing: 0,
	textDecorationLine: 'none',
	textTransform: 'none',
};
