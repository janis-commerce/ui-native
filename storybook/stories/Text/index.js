import React from 'react';
import TextComponent from '../../../src/components/Text';
import {StyleSheet} from 'react-native';

const Text = ({children, ...props}) => {
	const styles = StyleSheet.create({
		text: {
			...props,
		},
	});

	return <TextComponent style={styles.text}>{children}</TextComponent>;
};

export default Text;
