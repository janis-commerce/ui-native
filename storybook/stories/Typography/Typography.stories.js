import React from 'react';
import Typography from 'atoms/Typography';
import typography from 'theme/typography';

const types = Array.from(Object.keys(typography));

const sizes = ['large', 'medium', 'small', 'default'];

export default {
	title: 'Components/Typography',
	argTypes: {
		type: {
			options: types,
			control: {type: 'select'},
		},
		size: {
			options: sizes,
			control: {type: 'select'},
		},
		color: {
			control: {type: 'color'},
		},
	},
};

export const DefaultProps = ({textToDisplay, color, type, size}) => (
	<Typography color={color} type={type} size={size}>
		{textToDisplay}
	</Typography>
);

DefaultProps.storyName = 'default props';

DefaultProps.args = {
	textToDisplay: 'Hola',
	color: '#000000',
	type: 'display',
	size: 'medium',
};
