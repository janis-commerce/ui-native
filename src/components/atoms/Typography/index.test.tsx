import 'react-native';
import React from 'react';
import {create} from 'react-test-renderer';
import Text from './index';
import Typography from './index';

describe('Typography component', () => {
	describe('return error because children is undefined', () => {
		it('undefined children', () => {
			const {toJSON} = create(<Text />);
			expect(toJSON()).toBeNull();
		});
	});

	describe('render correct', () => {
		it('match snapshot with default typography', () => {
			const {toJSON} = create(<Typography>Janis Picking</Typography>);
			expect(toJSON()).toMatchSnapshot();
		});
	});
});
