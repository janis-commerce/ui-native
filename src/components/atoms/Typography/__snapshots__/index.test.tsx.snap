// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component render correct match snapshot with default typography 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "color": "#000000",
        "fontSize": 29,
        "fontWeight": "400",
        "lineHeight": 18,
      },
    ]
  }
>
  Janis Picking
</Text>
`;
